### DISCUSSION TITLE
Spacer inside Flex

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#2801] [General] [answered]
EDIT: The spacer works for row, just not column.
It's a bit frustrating that the Flex documentation has no examples of column.

From the examples in the documentation, it seems that Spacer inside Flex is equivalent to spacing:between.

But, in my use, it's not offering any spacing at all. In this screenshot, the container div has a red background.

![Chakra Flex Spacer](https://user-images.githubusercontent.com/55907892/102309536-3e368680-3f26-11eb-915e-9a01009b6fc3.png)

Here's my code:

```
        <Flex direction="column">
          <Box bg="blue.100">1</Box>
          <Spacer />
          <Box bg="blue.200">2</Box>
          <Spacer />
          <Box bg="blue.300">3</Box>
        </Flex>
```
Am I doing something wrong?

How do I get those Boxes to space evenly throughout the container?

**Comment by chr-ge** [comment, marked as answer]
You need to make the `Flex` container take up 100% height of the parent container:
```jsx
<Flex direction="column" height="100%">
  <Box bg="blue.100">1</Box>
  <Spacer />
  <Box bg="blue.200">2</Box>
  <Spacer />
  <Box bg="blue.300">3</Box>
</Flex>
```

**Reply by ghost** [reply]
Oh, awesome! Wow, there's so much I need to learn. Thanks so much!
